src/emc/kinematics/genhexkins.c

```diff
@@ -211,10 +221,10 @@ int genhexkins_read_hal_pins(void) {
     for (t = 0; t < NUM_STRUTS; t++) {
         b[t].x = haldata->basex[t];
         b[t].y = haldata->basey[t];
-        b[t].z = haldata->basez[t] + *haldata->tool_offset;
+        b[t].z = haldata->basez[t] + haldata->spindle_offset + *haldata->tool_offset;
```

Comment:
since spindle_offset is initialized as zero, this commit is a no-change for users who do not set spindle-offset. this is good.